src/niamoto/core/components/transforms/shapes.py

```
@@ -724,50 +726,86 @@ def _simplify_with_utm(
             geometry = (
                 gdf.geometry.union_all() if len(gdf) > 1 else gdf.geometry.iloc[0]
             )
+
+            # Simplify using UTM projection and convert to TopoJSON
             simplified = self._simplify_with_utm(geometry)
-            return mapping(simplified)
+            return self._convert_to_topojson(simplified)
 
         except Exception as e:
-            raise ValueError(f"Error simplifying geometry: {e}")
+            raise ValueError(f"Error processing geometry: {e}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)